packages/frontend/store/product/reducer.ts

```diff
@@ -2,16 +2,23 @@ import { createSlice } from '@reduxjs/toolkit';
 import type { HydrateAction } from '@types';
 import type { ProductDto } from '@vse-bude/shared';
 import { HYDRATE } from 'next-redux-wrapper';
-import { fetchProducts } from './actions';
+import {
+  addProductToFavorites,
+  deleteProductFromFavorites,
+  fetchProducts,
+  getFavoriteIds,
+} from './actions';
 
 interface ProductState {
   list: ProductDto[];
   loading: boolean;
+  favoriteProducts: string[];
 }
 
 const initialState: ProductState = {
   list: [],
   loading: false,
+  favoriteProducts: [],
```

Comment:
let's have a separate reducer - favoriteProducts. Since you have added useInFavorites we don't need to map favourites to products

Reply:
updated